src/main/java/kitchenpos/products/tobe/domain/vo/ProductName.java

```diff
@@ -0,0 +1,42 @@
+package kitchenpos.products.tobe.domain.vo;
+
+import java.util.Objects;
+import kitchenpos.common.purgomalum.PurgomalumClient;
+
+public class ProductName {
+    private final String value;
+
+    private ProductName(String value) {
+        this.value = value;
+    }
+
+    public static ProductName of(String value, PurgomalumClient purgomalumClient) {
+        if (Objects.isNull(value)) {
+            throw new IllegalArgumentException("Product name cannot be null");
+        } else if (purgomalumClient.containsProfanity(value)) {
+            throw new IllegalArgumentException("Product name contains profanity");
+        }
```

Comment:
굳이 if, else if 로 묶어서 처리하기보다는 별도의 if 문으로 처리해볼 수 있을 것 같네요. ([early return](https://www.youtube.com/shorts/UB0Jy6joK-8))

```suggestion
        if (Objects.isNull(value)) {
            throw new IllegalArgumentException("Product name cannot be null");
        } 
        
        if (purgomalumClient.containsProfanity(value)) {
            throw new IllegalArgumentException("Product name contains profanity");
        }
```